packages/sentinel/src/api/index.ts

```diff
@@ -55,11 +57,14 @@ export class SentinelClient extends BaseApiClient {
 
   public async update(sentinelId: string, sentinel: UpdateSentinelRequest): Promise<CreateSentinelResponse> {
     const currentSentinel = await this.get(sentinelId);
+
+    const updatedSentinel = await this.constructSentinelRequest({
+      ...this.toCreateSentinelRequest(currentSentinel),
+      ...(sentinel as CreateFortaSentinelRequest),
+    });
```

Comment:
the cast on line 63 should be `CreateSentinelRequest` but I dunno if casting here is ideal. `UpdateSentinelRequest` makes most properties in the sentinel model optional, meaning we could receive an object with just 1 property, which doesnt conform to `CreateSentinelRequest`. 

I think we should do something like this but maybe @spalladino has a better idea here
``` TS
    const updatedProperties = Object.keys(sentinel) as Array<keyof typeof sentinel>;
    for (const prop of updatedProperties) {
      (currentSentinel[prop] as any) = sentinel[prop];
    }

    const updatedSentinel = await this.constructSentinelRequest(currentSentinel);
```

It might be worthwhile adding some tests too to make sure we are updating the sentinel correctly

Reply:
Great you noticed!

I moved the type to be `CreateSentinelRequest`, but I went for the for-loop approach and the types just don't work. This API client has really weird type mixings and I tried many ways.

Fortunately, the casting is safe and I added a test for every function including the [case you're mentioning](https://github.com/OpenZeppelin/defender-client/pull/94/files#diff-f4c5281555d75ea82638ad7ec7c2023956929cb2978974c02020384d37be2f0fR269) in which we're not overriding